### PR TITLE
[WooCommerce Analytics] Add Common props, new events and bug fixing 

### PR DESCRIPTION
**Pull request by puntope**
## Proposed changes:

* Add a `session_id` as common prop
* Add `session_started` event
* Fix a bug: Fix `add_to_cart` event not being triggered in cart blocks
* Fix a bug: Fix `remove_from_cart` event not being triggered in classic cart page
* Fix a bug: Avoid tracking running in login page and ajax requests.
* Fix a bug: Avoid triggering `checkout_view` when visiting cart page.
* Fix a bug: Fix triggering cart_view doesn't work in  Blocks Cart
* Add common props as order_total, total_shipping, total_discount, total_tax, is_guest, timezone, store_currency and wp_version

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Prerequisites

1. Download and load Tracks Vigilante Chrome extension https://github.com/Automattic/tracks-chrome-extension
2. Install Woo and Jetpack with this PR
3. Connect Jetpack and WP.com
4. Create a page with Carts Block
5. Create a page with Cart classic
6. Create a page with Checkout Block
7. Create a page with CHeckout classic

* Testing session_started
1. Open an incognito page and navigate to shop
2. Verify `session_started` is tracked in tracks vigilante
3. Verify session_id is added as prop in this and future events

* Testing add_to_cart / remove from cart
5. Add some products to cart (no tracking is added in vigilante at this point)
8. Navigate to a product or refresh the page
9. See previous product_view + add_to_cart event  for previous additions
10. Go to the cart block page
11. See cart_view triggered
12. Remove an item and refresh
13. See remove_from_cart triggered
14. Go to shop again and add more products
15. Open mini cart and remove a product
16. Refresh. See remove_from_cart triggered
17. Go to Cart classic page
18. Remove a product 
19. See remove_from_cart triggered
20. Remove a product by reducing qty as 0 and clicking update
21. See remove_from_cart triggered

* Testing checkout view
23. Go to checkout classic
24. See product_checkout and checkout_view tracked with the right common props
25. Go to checkout blocks
26. See product_checkout and checkout_view tracked with the right common props
27. Complete the order
28. See order_confirmation_view and product_purchase triggered

* Close the incognito window and open a new one

29. Enter the login page
30. Nothing is tracked
31. Enter to Shop in the browser 
32. A new session_started is loaded

### CODE REVIEW

**Comment by ianlin** [comment]
@puntope Thanks for adding more props to woocommerce analytics package.

> 2. Install Woo and Jetpack with this PR

I guess this means install the `woocommerce-analytics` package using this PR in WC. For building the `woocommerce-analytics` package I followed the below two links:

- https://github.com/Automattic/jetpack/blob/trunk/docs/quick-start.md
- https://github.com/Automattic/jetpack/blob/trunk/docs/development-environment.md

The command `jetpack build` brought me into an interactive CLI and it seems the package was built successfully:

Screenshot 2024-12-12 at 20 19 10

Then I tried to edit WC's `composer.json` and add the below config under the `repository` section:

```json
{
    "type": "path",
    "url": "/path/to/jetpack/projects/packages/woocommerce-analytics",
    "options": {
        "symlink": false
    }
}
```

and rebuilt WC, but the package did not seem to be installed in the `vendor/` directory. I also ran `composer install` manually but it was the same.

Did I do it correctly or if there is something that I missed?

**Comment by jeherve** [comment]
@ianlin Since the package is still required by the Jetpack plugin, if you activate the Jetpack plugin in your test environment, you won't have to do anything else. You won't need to start requiring the package anywhere else.

**Comment by puntope** [comment]
> @ianlin Since the package is still required by the Jetpack plugin, if you activate the Jetpack plugin in your test environment, you won't have to do anything else. You won't need to start requiring the package anywhere else.

> Install Woo and Jetpack with this PR

Correct. Thanks @jeherve for assisting. 

The only thing to do is to build Jetpack and the package will be auto-loaded and initialized by  https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/woocommerce-analytics.php

**Comment by puntope** [comment]
btw @jeherve Forgot to add you as a reviewer as well. Not necessary and totally optional as I don't want to load you of extra work. But welcome if you wanna take a look.

**Comment by ianlin** [comment]
Thanks for the heads up, I successfully ran Jetpack in the docker with this PR.

Not sure if I missed something, it seems that the `session_started` event was not fired in my environment, however, the `product_view` + `add_to_cart` events were fired with the newly added common props like `session_id` and `order_value`, that implies the environment was running with this PR correctly.


https://github.com/user-attachments/assets/4034ceb2-0aa4-4d74-90fd-d00decdbca5e

**Comment by puntope** [comment]
> it seems that the session_started event was not fired in my environment

That event runs even before entering the shop. Also notice that if you're in the same session then the event is not sent because you're in the same session.

Can you send me your `_ui` prop? So I can check in Tracks.

**Comment by ianlin** [comment]
> That event runs even before entering the shop. Also notice that if you're in the same session then the event is not sent because you're in the same session.
>
> Can you send me your _ui prop? So I can check in Tracks.

Got it. I reopen an incognito page and go to the shop page, add a product to cart then go to cart page, these are the events for `product_view` and `add_to_cart`

Screenshot 2024-12-16 at 13 05 49

The value of `_ui` is `VqWRkYjvBBZxyVvE0uNgPwGU`. The event was fire at 04:58:36 UTC.

On Tracks I can search `product_view` and add_to_cart events by my `_ui`, but there is no search result for `session_started`.

**product_view**

Screenshot 2024-12-16 at 13 15 00

**add_to_cart**

Screenshot 2024-12-16 at 13 15 07

**session_started**

Screenshot 2024-12-16 at 13 15 14

**Comment by ianlin** [comment]
Hey @puntope,

I found it's because my Brave browser blocks scripts by default, and I disable it after the page is loaded, but the `session_started` event was already blocked. I disable the script blocking globally and right now I can see the `session_started` event being sent.

**Comment by ianlin** [comment]
I've completed the testing flow you provided, and I found there were two things seem not working as expected:

> 23. Go to checkout classic
> 24. See product_checkout and checkout_view tracked with the right common props

It probably isn't related to this PR. I added three products to the cart then went to checkout classic, three `product_checkout` events was tracked, but the product id prop `pi` are identical in all events.


https://github.com/user-attachments/assets/de6d256a-f7c2-498b-980f-0bd025086de5

> 29. Enter the login page
> 30. Nothing is tracked

I opened a new incognito page and went to login page by entering the URL `https://my-site/my-account/`, but the `session_started` event was still sent.


https://github.com/user-attachments/assets/380c9b75-e518-483c-8fca-6e55c71b48f6

**Comment by puntope** [comment]
> I opened a new incognito page and went to login page by entering the URL `https://my-site/my-account/`, but the `session_started` event was still sent.

This is expected. Session Started runs on every page (Not admin)

**Comment by puntope** [comment]
Howdy @ianlin Thanks for the review. I fixed the pending bug regarding the pi being the same in all the events for `checkout_product_view` This is ready for a new round.

File: projects/packages/woocommerce-analytics/changelog/add-common-props [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: changed
+
+Add common props, more events and bug fixing.
```
Comment (ianlin) [inline projects/packages/woocommerce-analytics/changelog/add-common-props:4]: We could probably make the changelog more specific to `woocommerce-analytics` package.

**Review by ianlin — APPROVED** [review]
Thanks for adding common props and new events, as well as fixing some bugs. I tested all the scenarios I can think of and they work expected.

Just one thing I'm not so sure but maybe we can make it a follow up PR:

The props:
- `cart_page_contains_cart_block`
- `cart_page_contains_cart_shortcode`
- `checkout_page_contains_checkout_block`
- `checkout_page_contains_checkout_shortcode`

Although these props appear in all the events, but I think they are only relevant to the events `cart_view` and `checkout_view`. However, when I went to **Cart Classic** page the value seems to be incorrect, I expected it to be:

- `cart_page_contains_cart_block`: 0
- `cart_page_contains_cart_shortcode`: 1

But it actually gave me:

- `cart_page_contains_cart_block`: 1
- `cart_page_contains_cart_shortcode`: 0

For **Checkout Classic** page it was the same.

I guess those props are not really important for our requirements for now, so I'm approving this PR now.

File: projects/packages/woocommerce-analytics/changelog/add-common-props (hunk 1 above)
Comment (puntope) [reply projects/packages/woocommerce-analytics/changelog/add-common-props:4]: The changelog is inside woocommerce-analytics package folder. So I believe that scopes everything inside the folder to be woocommerce-analytics specific?

**Comment by puntope** [comment]
> I guess those props are not really important for our requirements for now, so I'm approving this PR now.

PErfect. I will investigate in a follow up PR

**Pull request merged**